drivers/interrupt_controller/wuc_ite_it8xxx2.c

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright (c) 2022 ITE Corporation. All Rights Reserved
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <device.h>
+#include <drivers/interrupt_controller/wuc_ite_it8xxx2.h>
+#include <dt-bindings/interrupt-controller/it8xxx2-wuc.h>
+#include <kernel.h>
+#include <soc.h>
+#include <soc_common.h>
+#include <stdlib.h>
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(wuc_ite_it8xxx2, LOG_LEVEL_ERR);
+
+#define DT_DRV_COMPAT ite_it8xxx2_wuc
+
+/* Driver config */
+struct it8xxx2_wuc_cfg {
+	/* WUC wakeup edge mode register */
+	uintptr_t reg_wuemr;
+	/* WUC wakeup edge sense register */
+	uintptr_t reg_wuesr;
+	/* WUC wakeup enable register */
+	uintptr_t reg_wuenr;
+	/* WUC wakeup both edge mode register */
+	uintptr_t reg_wubemr;
+};
+
+void it8xxx2_wuc_enable(const struct device *dev, uint8_t mask)
+{
+	const struct it8xxx2_wuc_cfg *config = dev->config;
+	volatile uint8_t *reg_wuenr = (uint8_t *)config->reg_wuenr;
```

Comment:
why not make `reg_wuenr` directly a pointer?

Reply:
Done